### PR TITLE
FIX: it seems that fabio doesn't flip the masks on the way out so we …

### PR DESCRIPTION
**Pull request by CJ-Wright**
…shall

### CODE REVIEW

File: skbeam/io/fit2d.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def fit2d_save(mask, filename, dir_path=None):
     dir_path: str, optional
         Path to the destination file
     """
-    saver = Fit2dMaskImage(data=~mask)
+    saver = Fit2dMaskImage(data=~np.flipud(mask))
```
Comment (licode) [inline skbeam/io/fit2d.py]: Please put a note in function doc to mention this. It looks like a fix is needed in fabio?

**Comment by licode** [comment]
I just fixed some bugs in tests, which can now properly cover all the functions. Please rebase this PR and send it again.

**Comment by licode** [comment]
I also find a issue relate to fit2d_save, https://github.com/scikit-beam/scikit-beam/blob/master/skbeam/io/tests/test_fit2d_save.py#L49

I put it as xfail. Do you have any input on this? thanks.

**Comment by licode** [comment]
any update here?

**Comment by CJ-Wright** [comment]
Sorry for the delay I just need to check this against fit2d.

File: skbeam/io/fit2d.py (hunk 1 above)
Comment (ordirules) [reply skbeam/io/fit2d.py]: yes explanation would help? why is this needed?

**Comment by CJ-Wright** [comment]
See https://github.com/silx-kit/fabio/issues/161

File: skbeam/io/fit2d.py [hunk 2]
```diff
@@ -35,5 +35,5 @@ def read_fit2d_msk(filename):
     ndarray:
         The mask as boolian array
     """
-    a = fabio.open(filename)
+    a = np.flipud(fabio.open(filename))
```
Comment (ordirules) [inline skbeam/io/fit2d.py]: this should not be flipped?

**Comment by ordirules** [comment]
If I understand correctly, based on that thread, the saving needs to be flipped, but in principle the reading should not need to be. Can you confirm this @CJ-Wright ? (left a comment on the line to precise the position)

**Comment by ordirules** [comment]
(also we should add a test to validate this)

**Comment by ericdill** [comment]
@CJ-Wright -- no longer of interest?

**Comment by CJ-Wright** [comment]
I think that the fabio issue has the correct answer. If things are to be fixed they should be fixed there.

**Pull request closed without merging**